src/group_theory/monoid_localization.lean

```diff
@@ -147,37 +158,56 @@ namespace localization_map
 
 variables {S N}
 
-@[to_additive, ext] lemma ext {f g : localization_map S N} (h : ∀ x, f.1 x = g.1 x) : f = g :=
-by cases f; cases g; simp only []; exact monoid_hom.ext h
+/-- Makes a localization map from a `comm_monoid` hom satisfying the characteristic predicate. -/
+@[to_additive "Makes a localization map from an `add_comm_monoid` hom satisfying the characteristic predicate."]
+def of_monoid_hom (f : M →* N) (H1 : ∀ y : S, is_unit (f y))
```

Comment:
If you call this `monoid_hom.to_localization_map` and move outside of `namespace localization`, you'll be able to use dot notation. Don't know whether you want this.

Reply:
good point! Changing this